server/src/test/java/io/spine/examples/shareaware/server/e2e/given/E2EUser.java

```diff
@@ -0,0 +1,277 @@
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.e2e.given;
+
+import com.google.common.collect.ImmutableList;
+import io.spine.base.CommandMessage;
+import io.spine.base.EntityState;
+import io.spine.base.EventMessage;
+import io.spine.client.Client;
+import io.spine.client.Subscription;
+import io.spine.core.UserId;
+import io.spine.examples.shareaware.WalletId;
+import io.spine.examples.shareaware.investment.InvestmentView;
+import io.spine.examples.shareaware.investment.command.PurchaseShares;
+import io.spine.examples.shareaware.market.AvailableMarketShares;
+import io.spine.examples.shareaware.server.given.GivenWallet;
+import io.spine.examples.shareaware.share.Share;
+import io.spine.examples.shareaware.wallet.WalletBalance;
+import io.spine.examples.shareaware.wallet.command.CreateWallet;
+import io.spine.examples.shareaware.wallet.command.ReplenishWallet;
+import io.spine.examples.shareaware.wallet.command.WithdrawMoney;
+import io.spine.money.Money;
+import io.spine.testing.core.given.GivenUserId;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.truth.Truth.*;
+import static io.spine.examples.shareaware.server.e2e.given.E2EUserTestEnv.purchaseSharesFor;
+import static io.spine.examples.shareaware.server.e2e.given.E2EUserTestEnv.replenishWallet;
+import static io.spine.examples.shareaware.server.e2e.given.E2EUserTestEnv.withdrawMoneyFrom;
+import static io.spine.examples.shareaware.server.given.GivenWallet.createWallet;
+import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Represents a user for end-to-end tests which interacts
+ * with the server-side with the help of {@link Client}.
+ *
+ * <p>In some tests, it may be required to extend the API provided by this type.
+ * Then, subclassing should be done in a nested type, residing inside
+ * the target test class.
+ */
+public class E2EUser {
```

Comment:
Looking at all the methods in this class, I think we might not even need `SharePurchaseUser` descendant. We can just extend the API of this type, and that's it (making this type `final` and adjusting the documentation).